packages/scaffold/src/views/w3m-approve-transaction-view/index.ts

```diff
@@ -20,34 +30,34 @@ export class W3mApproveTransactionView extends LitElement {
 
   public constructor() {
     super()
+
     this.unsubscribe.push(
-      ModalController.subscribeKey('open', val => {
-        if (!val) {
-          this.onHideIframe()
-        }
-      })
+      ...[
+        ModalController.subscribeKey('open', isOpen => {
+          if (!isOpen) {
+            this.onHideIframe()
+            RouterController.popTransactionSuccessAction()
```

Comment:
Double check if there is a state that approve transaction page is triggered but the transaction success stack is not updated.